sotodlib/site_pipeline/preprocess_tod.py

```diff
@@ -152,12 +154,12 @@ def load_preprocess_det_select(obs_id, configs, context=None):
     """
     configs, context = _get_preprocess_context(configs, context)
     
-    pipe = Pipeline(configs["process_pipe"], logger=logger)
     meta = context.get_meta(obs_id)
-
-    for process in pipe:
-        logger.info(f"Selecting On {process.name}")
-        process.select(meta)
+    proc_aman = meta["preprocess"]
```

Comment:
@kmharrington this is how I chose to select on the final process valid flags. I would prefer to get `assn` from `configs["process_pipe"]` however the name that is wrapped in is an optional argument in each process + not always defined consistently and using the process.name doesn't work because that's not unique (i.e. you can call the same process with the same name multiple times in the same pipe) so I used the order in aman.preprocess._assignments because the order is the order they're wrapped in the pipe.

Reply:
Honestly this looks pretty clunky and possible prone to failure. Pipelines are lists, why not just run `pipe[-1].select`?

Reply:
I'll try it.

Reply:
Ok @kmharrington I've fixed this in [c702f74](https://github.com/simonsobs/sotodlib/pull/751/commits/c702f744b852b63843840255c6c6baca94e5c401)

Reply:
ya looks good!